themes/bootstrap3/js/search.js

```diff
@@ -0,0 +1,371 @@
+/*global VuFind */
+
+VuFind.register('search', function search() {
+  let jsRecordListSelector = '.js-result-list';
+  let paginationLinksSelector = '.js-pagination a,.js-pagination-simple a';
+  let scrollElementSelector = '.search-stats';
+  let searchStatsSelector = '.js-search-stats';
+  let searchFormSelector = 'form.searchForm';
+  let resultsControlFormSelector = '.search-controls form';
+  let sortFormSelector = resultsControlFormSelector + '.search-sort';
+  let limitFormSelector = resultsControlFormSelector + '.search-result-limit';
+  let viewTypeSelector = '.view-buttons a';
+
+  // Forward declaration
+  let loadResults = function loadResultsForward() {};
+
+  /**
+   * Get the URL without any parameters
+   *
+   * @param {string} url
+   *
+   * @returns string
+   */
+  function getBaseUrl(url) {
+    const parts = url.split('?');
+    return parts[0];
+  }
+
+  /**
+   * Initialize pagination.
+   */
+  function initPagination() {
+    document.querySelectorAll(paginationLinksSelector).forEach((element) => {
+      if (!element.dataset.ajaxPagination) {
+        element.dataset.ajaxPagination = true;
+        element.addEventListener('click', function handleClick(event) {
+          event.preventDefault();
+          const href = this.getAttribute('href');
+          loadResults(href, true);
+        });
+      }
+    });
+  }
+
+  /**
+   * Initialize result controls that are not refreshed via AJAX.
+   *
+   * Note that view type links are updated in updateResultControls, but using them
+   * will cause a reload since page contents may change.
+   */
+  function initResultControls() {
+    document.querySelectorAll(resultsControlFormSelector).forEach((form) => {
+      if (!form.dataset.ajaxPagination) {
+        form.dataset.ajaxPagination = true;
+        form.querySelectorAll('.jumpMenu').forEach(jumpMenu => {
```

Comment:
This can adapt to whatever #2814 brings, but with this the need to get rid of jump menus due to accessibility issues vanishes. Since this can keep the focus in the control that was updated, there's no sudden and unexpected jump that causes accessibility issues. #2492 and #2814 were actually the ones that got me started working on this one (along the other benefits for sure, but still).

Reply:
I guess my question is: if this new logic supersedes the setupJumpMenus function, can we just get rid of that function entirely? We then presumably need to display submit buttons in cases where the AJAX search results are disabled. This should already work for the case where JS is fully disabled, but I'm wondering if additional work might be needed for the case where AJAX search results are disabled via configuration.

If we get to a place where we display submit buttons when needed and have accessible behavior in AJAX mode, then we might not need #2814 at all (or we might be able to start a new PR that takes some smaller parts without requiring so much additional work to be done).

And then that brings me back to the question: is it worth doing any of that here, or should we wait and do it as follow-up?

Reply:
Right, I missed that part of the idea. Yes, that'd make sense. I'd prefer to do it in a follow-up as this is already big enough, but if you think it's better to incorporate it here to keep all the major changes in one set, I can mange that as well.

Reply:
I don't mind waiting!